lib/SILOptimizer/Utils/LexicalDestroyHoisting.cpp

```diff
@@ -193,26 +189,15 @@ bool Dataflow::classificationIsBarrier(Classification classification) {
   llvm_unreachable("exhaustive switch not exhaustive?!");
 }
 
-void Dataflow::visitedInstruction(SILInstruction *instruction,
-                                  Classification classification) {
-  assert(classifyInstruction(instruction) == classification);
-  switch (classification) {
-  case Classification::Barrier:
-    result.barriers.push_back(instruction);
-    return;
-  case Classification::Other:
-    return;
-  }
-  llvm_unreachable("exhaustive switch not exhaustive?!");
-}
-
-bool Dataflow::checkReachableBarrier(SILInstruction *instruction) {
+Dataflow::Effect Dataflow::effectForInstruction(SILInstruction *instruction) {
+  if (llvm::find(uses.ends, instruction) != uses.ends.end())
```

Comment:
It looks like calling `effectForInstruction` for each instruction is now quadratic.

Reply:
Fixed this issue in LexicalDestroyHoisting, ShrinkBorrowScope, and SSADestroyHoisting: https://github.com/apple/swift/pull/59057